From 1e0e888ec56bf23f38c8012172bc854a7194ce49 Mon Sep 17 00:00:00 2001
From: Negreanu Marius Adrian <adrian.m.negreanu@intel.com>
Date: Thu, 6 Oct 2011 11:17:03 +0300
Subject: gdbclient: return if executable was not found

Problem:
gdb might report an error, when it doesnt find the file
but still continue to display the prompt. This may be
overlooked and considered an "ok" situation.

Fix:
check if the executable exists *before* starting gdb.
This way the noise generated by gdb is avoided

Change-Id: I226944d8fc9c146403ccef23ac6c5772935ac700
---
 envsetup.sh |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/envsetup.sh b/envsetup.sh
index cbc1371..93a4ab7 100644
--- a/envsetup.sh
+++ b/envsetup.sh
@@ -856,6 +856,10 @@ function gdbclient()
        else
            EXE="app_process"
        fi
+       if [ ! -f "$OUT_EXE_SYMBOLS/$EXE" ]; then
+           echo "Unable to find executable: $OUT_EXE_SYMBOLS/$EXE"
+           return 1
+       fi
 
        local PORT="$2"
        if [ "$PORT" ] ; then
-- 
1.7.4.1

